### PR TITLE
[HOLD] manually merge setzors fishing code

### PR DESCRIPTION
**Pull request by godmode**
### Testing
I got this running by testing at Lake Tepokalipuka in East Sarutabaruta with a Judge's Rod and a Robber Rig. I mostly fished up fish this way. Monsters come more easily with Fly Lure here.
```
!pos -250 0 -126 116
!additem 17012
!additem 17002
!additem 17405
``` 

### Deviations from original code
To avoid the merge hell of about a year of missing commits, I did a compare on Setzor's code from the DSP version it was based off of. Then, on a separate repo I pulled the latest commit of master (as of 10/12/2018) and made changes to it as needed. There were only a few things that were changed from Setzor's original code.
1. The `caught_fish` class was removed in a later commit so I added that back in and updated the packet structure.
2. Upon fishing a mob that is not loaded (ex. all positions 0) the game would crash. I added a pointer guard. I just made it print a log and send the player a fishing end packet.
3. There were runtime warnings booting the server because settings were in the map config file but not elsewhere--I added those. Although none of the fishing settings I've been able to tell are used right now I think Setzor's intention was to take advantage of them.

### Contributing and getting this merged!
- Largely, the database needs to be filled. It can be guesstimates or even scripted to use the current system for now and updated as people get more accurate data though. I will update this with the DB structure in the future. I didn't check much into different places to fish but there aren't many right now that will yield results. I just found Lake Tepokalipuka querying the DB so I stuck there to get it going.
- Obviously, this was still WIP so there are a lot of prints everywhere and debugging methods. I haven't used the commented out debugging methods in `lua_baseentity` yet so if someone else wanted to play around with that, knock yourselves out. I didn't need to look at it yet.
- Rod/line breaking
- Golden arrow calculation
- regen/drain formula based on skill level

### Potentially useful debugging SQL
```sql
-- Select all fish/items from a fishing point
SELECT 
distinct  
ff.fishid,                
ff.name,                  
ff.min_skill_level,       
ff.skill_level,           
ff.size,                  
ff.base_delay,            
ff.base_move,             
ff.min_length,            
ff.max_length,            
ff.size_type,             
ff.water_type,            
ff.log,                   
ff.quest,                 
ff.item,                  
ff.max_hook,              
ff.required_keyitem,      
ff.required_catches,      
fla.power                 
FROM fishing_lure_affinity fla 
LEFT JOIN fishing_fish ff 
ON ff.fishid = fla.fishid 
WHERE(fla.lureid = 17002 || ff.item = 1) 
AND fla.fishid IN(select fg.fishid FROM fishing_group fg WHERE fg.groupid = (select groupid from fishing_catch where zoneid = 116 and areaid = 5)) 
GROUP BY ff.fishid;

-- Select all mobs from a fishing zone
SELECT 
mobid,                
name,                 
level,                
size,                 
base_delay,           
base_move,            
log,                  
quest,                
nm,                   
rarity,               
required_key_item     
FROM fishing_mob 
WHERE zoneid = 116;
```

### CODE REVIEW

**Comment by zynjec** [comment]
This should probably get its own new branch and merged there, as it's going to need a lot more than one PR to get this ready for master.

**Comment by TeoTwawki** [comment]
Made it so. We actually had a fishing branch but its old af now so I just made a new one off current master

File: scripts/globals/fishing/fishing.lua [hunk 1]
```diff
@@ -0,0 +1,243 @@
+-------------------------------------------------
+--	Author: Setzor : setzor@gmail.com
```
[the remark this answers is not in the document]
Comment (TeoTwawki) [reply scripts/globals/fishing/fishing.lua]: Prior instances of this have been being removed so safe to say this is. It wanted. Credit belongs in commit messages.

**Comment by godmode** [comment]
Just had a chat with Setzor. Force pushed him as the author of the initial commit. Other changes will be done later but branch ready to be pulled and submitted to now.

**Comment by zynjec** [comment]
No real feedback/movement with the minor changes wanted re: packets.

Feel free to PR this again once it's completed.

**Pull request closed without merging**